src/machinelist.cpp

```diff
@@ -892,14 +892,11 @@ void MachineList::verify(bool currentOnly)
 	verifyProc->start(command, args, QIODevice::ReadOnly | QIODevice::Text);
 }
 
-QString MachineList::value(QString element, QString attribute, bool translate)
+QString MachineList::value(QXmlStreamReader *element, QString attribute, bool translate)
 {
-	QString attributePattern(" " + attribute + "=\"");
-	if ( element.contains(attributePattern) ) {
-		QString valueString(element.remove(0, element.indexOf(attributePattern) + attributePattern.length()));
-		valueString = valueString.remove(valueString.indexOf("\""), valueString.lastIndexOf(">")).replace("&amp;", "&").replace("&lt;", "<").replace("&gt;", ">").replace("&quot;", "\"").replace("&apos;", "'");
-		if ( valueString == ">" )
-			return QString();
+	if ( element->attributes().hasAttribute(attribute) ) {
+		QString valueString(element->attributes().value(attribute).toString());
+		valueString = valueString.replace("&amp;", "&").replace("&lt;", "<").replace("&gt;", ">").replace("&quot;", "\"").replace("&apos;", "'");
```

Comment:
Another place you don’t need to do XML character entity decoding.

Reply:
Done